compendium/ServiceComponent/test/TestComponentInstance.cpp

```diff
@@ -288,6 +325,19 @@ namespace
         ASSERT_FALSE(compObj->GetBar());
     }
 
+    TEST(ComponentInstance, validateConstructorCall)
```

Comment:
I am not sure. Originally that is the way I was structuring it but to me, it looks like all other tests in the serviceComponent folder are failure tests (ie they all fail to compile). In contrast, we are verifying non-failure. I was getting a problem with multiple definitions of the main function which I think were because my test case was compiling and getting confused with the TestComponentInstance.cpp main file. 

Given that we are verifying that it does compile, I figured putting it in this folder was most appropriate.